traffic_ops/testing/api/v14/roles_test.go

```diff
@@ -105,6 +106,34 @@ func GetTestRoles(t *testing.T) {
 
 }
 
+func VerifyGetRolesOrder(t *testing.T) {
+	role := testData.Roles[roleGood]
+	params := map[string]string{
+		"name":      *role.Name,
+		"orderby":   "name",
+		"sortOrder": "desc",
+	}
+	descResp, _, status, err := TOSession.GetRoleByQueryParams(params)
+	log.Debugln("Status Code: ", status)
+	if err != nil {
+		t.Errorf("cannot GET Role by role: %v - %v\n", err, descResp)
+	}
+	params["sortOrder"] = "asc"
+	ascResp, _, status, err := TOSession.GetRoleByQueryParams(params)
+	log.Debugln("Status Code: ", status)
+	if err != nil {
+		t.Errorf("cannot GET Role by role: %v - %v\n", err, ascResp)
+	}
+
+	// reverse the descending-sorted response and compare it to the ascending-sorted one
+	for start, end := 0, len(descResp)-1; start < end; start, end = start+1, end-1 {
+		descResp[start], descResp[end] = descResp[end], descResp[start]
+	}
+	if !reflect.DeepEqual(descResp, ascResp) {
```

Comment:
I tested this with the sortOrder parameter not implemented and it doesn't fail, so something is definitely wrong but I can't seem to figure out what

Reply:
if sortOrder isn't implemented wouldn't descResp be exactly ascResp?

Reply:
Yep, but if descResp is then reversed, they shouldn't be equal. And yet the test passes anyway...

Reply:
Oh, I misunderstood.

Reply:
Fixed it. Shouldn't have had the `name` in the query parameters, which was limiting the server response size to one